src/distribution/Airdrop.sol

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.24;
+
+import { IERC20, SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import { MerkleProof } from "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+import { EnumerableSet } from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+
+import { IAirdrop } from "./interfaces/IAirdrop.sol";
+
+/// @title Airdrop
+/// @notice A contract for distributing token airdrops using a merkle tree
+/// @dev Inspired by Uniswap https://github.com/Uniswap/merkle-distributor/blob/master/contracts/MerkleDistributor.sol at commit 25a79e8
+contract Airdrop is IAirdrop, Ownable {
+    using SafeERC20 for IERC20;
+    using EnumerableSet for EnumerableSet.Bytes32Set;
+
+    mapping(bytes32 merkleRoot => address token) public distributions;
+    mapping(bytes32 merkleRoot => mapping(uint256 index => uint256 claimed)) public claimed;
+    EnumerableSet.Bytes32Set private merkleRoots;
+
+    constructor() Ownable(msg.sender) {}
+
+    /// @inheritdoc IAirdrop
+    function addDistributions(address token, bytes32 merkleRoot) external override onlyOwner {
+        if (merkleRoots.contains(merkleRoot)) revert DistributionAlreadyExists();
+        distributions[merkleRoot] = token;
+        merkleRoots.add(merkleRoot);
+
+        emit DistributionAdded(token, merkleRoot);
+    }
+
+    /// @inheritdoc IAirdrop
+    function claim(
+        address account,
+        uint256[] calldata index,
+        uint256[] calldata amount,
+        bytes32[][] calldata merkleProof,
+        bytes32[] calldata merkleRoot
+    ) public override {
+        for (uint256 i = 0; i < index.length; i++) {
+            if (isClaimed(index[i], merkleRoot[i])) revert AlreadyClaimed();
+            address token = distributions[merkleRoot[i]];
+
+            // Verify the merkle proof.
+            bytes32 node = keccak256(abi.encodePacked(index[i], account, amount[i]));
+            if (!MerkleProof.verify(merkleProof[i], merkleRoot[i], node)) revert InvalidProof();
+
+            // Mark it claimed and send the token.
+            _setClaimed(index[i], merkleRoot[i]);
+            IERC20(token).safeTransfer(account, amount[i]);
```

Comment:
I think `token` is only used here, can you make this:

`IERC20(distributions[merkleRoot[i]]).safeTransfer(account, amount[i]);`

and then remove that line above.